app/src/test/java/org/mozilla/fenix/perf/PerformanceInflaterTest.kt

```diff
@@ -0,0 +1,82 @@
+package org.mozilla.fenix.perf
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.FrameLayout
+import io.mockk.every
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+import java.io.File
+
+@RunWith(FenixRobolectricTestRunner::class)
+class PerformanceInflaterTest {
+
+    private lateinit var perfInflater : MockInflater
+
+    private val layoutNotToTest = setOf(
+        "fragment_browser",
+        "fragment_add_on_internal_settings"
+    )
+
+    private val layoutWithMerge = setOf(
+        "mozac_ui_tabcounter_layout",
+        "tabstray_multiselect_items",
+        "tab_preview",
+        "tracking_protection_category"
+    )
+
+    @Before
+    fun setup() {
+        InflationCounter.inflationCount.set(0)
+
+        perfInflater = MockInflater(LayoutInflater.from(testContext), testContext)
+    }
+
+    @Test
+    fun `WHEN we inflate a view, the inflation counter should increase`() {
+        assertEquals(0, InflationCounter.inflationCount.get())
+        perfInflater.inflate(R.layout.fragment_home, null, false)
+        assertEquals(1, InflationCounter.inflationCount.get())
+    }
+
+    @Test
+    fun `WHEN inflating one of our resource file, the inflater should not crash`(){
```

Comment:
This technically tests robolectric's implementation rather than the android framework's – I wonder if we should make this an on-device test rather than a unit test. They're pretty fast when there's no UI to display. Would that also help remove your `<fragment>` hacks?

Reply:
Good question about the `<fragment>` hack being solve if we use a on-device test. I was wondering if this would work for on device test since I'm manually retrieving all the files from a specific file path. I thought I wouldn't be able to do so because I assumed it kinda acted as an `.apk`